diesel/src/pg/expression/extensions/tablesample_dsl.rs

```diff
@@ -0,0 +1,71 @@
+use crate::pg::query_builder::tablesample::{BernoulliMethod, SystemMethod};
+use crate::query_builder::Tablesample;
+use crate::Table;
+
+/// The `tablesample` method
+///
+/// The `TABLESAMPLE` clause is used to select a randomly sampled subset of rows from a table.
+///
+/// This is only implemented for the Postgres backend. While `TABLESAMPLE` is standardized in
+/// SQL:2003, in practice each RDBMS seems to implement a superset of the SQL:2003 syntax,
+/// supporting a wide variety of sampling methods.
+///
+/// Calling this function on a table (`mytable.tablesample(...)`) will result in the SQL
+/// `FROM mytable TABLESAMPLE ...` --
+/// `mytable.tablesample(...)` can be used just like any table in diesel since it implements
+/// [Table](crate::Table).
+///
+/// The `BernoulliMethod` and `SystemMethod` types can be used to indicate the sampling method for
+/// a `TABLESAMPLE method(p)` clause where p is specified by the portion argument. The provided
+/// percentage should be an integer between 0 and 100.
+///
+/// If the seed argument is is Some(f) then f becomes the seed in `TABLESAMPLE ... REPEATABLE (f)`.
```

Comment:
This part is not correct anymore, it should likely be moved to the `with_seed()` function. We likely want to link that function from here as well?